web/core/store/issue/issue-details/attachment.store.ts

```diff
@@ -105,20 +126,51 @@ export class IssueAttachmentStore implements IIssueAttachmentStore {
   };
 
   createAttachment = async (workspaceSlug: string, projectId: string, issueId: string, file: File) => {
-    const response = await this.issueAttachmentService.uploadIssueAttachment(workspaceSlug, projectId, issueId, file);
-    const issueAttachmentsCount = this.getAttachmentsByIssueId(issueId)?.length ?? 0;
-
-    if (response && response.id) {
+    const tempId = uuidv4();
+    try {
+      // update attachment upload status
       runInAction(() => {
-        update(this.attachments, [issueId], (attachmentIds = []) => uniq(concat(attachmentIds, [response.id])));
-        set(this.attachmentMap, response.id, response);
-        this.rootIssueStore.issues.updateIssue(issueId, {
-          attachment_count: issueAttachmentsCount + 1, // increment attachment count
+        set(this.attachmentsUploadStatusMap, [issueId, tempId], {
+          id: tempId,
+          name: file.name,
+          progress: 0,
+          size: file.size,
+          type: file.type,
         });
       });
-    }
+      const response = await this.issueAttachmentService.uploadIssueAttachment(
+        workspaceSlug,
+        projectId,
+        issueId,
+        file,
+        (progressEvent) => {
+          runInAction(() => {
```

Comment:
This will get triggered too often and will  create perf problems when we support multiple file uploads. Lets batch and throttle this updates.